File: src/main/java/no/sws/invoice/recipient/Recipient.java
/*
 * Copyright (C) 2009 Pal Orby, SendRegning AS. <http://www.balder.no/> This program is free software: you can
 * redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later version. This program is distributed in
 * the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details. You should have received a
 * copy of the GNU General Public License along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package no.sws.invoice.recipient;

import java.util.List;

/**
 * @author Pal Orby, SendRegning AS
 */
public interface Recipient {

    public static final Recipient NOT_FOUND = new RecipientImpl();

    public String getRecipientNo();

	public void setRecipientNo(String recipientNo);

	public String getName();

	/**
	 * Set name.
	 *
	 * @param name to set.
	 * @throws IllegalArgumentException If name is null or zero length.
	 */
	public void setName(String name);

	public String getAddress1();

	public void setAddress1(String address1);

	public String getAddress2();

	public void setAddress2(String address2);

	public String getZip();

	/**
	 * Set zip.
	 *
	 * @param zip to set.
	 * @throws IllegalArgumentException If zip is null or zero length.
	 */
	public void setZip(String zip);

	public String getCity();

	/**
	 * Set city.
	 *
	 * @param city to set.
	 * @throws IllegalArgumentException If city is null or zero length.
	 */
	public void setCity(String city);

	public String getCountry();

	public void setCountry(String country);

	public String getEmail();

	public void setEmail(String email);

	public void setPhone(String phone);

	public void setMobile(String mobile);

	public void setFax(String fax);

	public void setWeb(String web);

	public void setCategory(String category);

    public String getCategory();

	public void setComment(String comment);

	public void setCreditDays(String creditDays);

	public void setOrgNo(String orgNo);

	public void setPreferredShipment(String preferredShipment);

	public void setAttachPdf(String attachPdf);

    public List<String> getCategories();

    public void setCategories(List<String> categories);

    public String getOrgNo();

    public String getMobile();

    public String getPhone();

    public String getFax();
}

File: trunk/src/main/python/sws/recipient/recipient.py
#! /usr/bin/env python

class Recipient:
	''' The class that contains all information
	about a recipient. It contains get and set 
	methods for every information variable '''
	def __init__(self):
		self.init = "true"		
	def get_name(self):
		return self.name	
	def get_zipcode(self):
		return self.zipcode
	def get_city(self):
		return self.city
	def get_address1(self):
		return self.address1
        def get_address2(self):
                return self.address2
	def get_country(self):
		return self.country
	def get_recipientNo(self):
		return self.recipientNo
	def get_phone(self):
		return self.phone
	def get_mobile(self):
		return self.mobile
	def get_fax(self):
		return self.fax
	def get_email(self):
		return self.email
	def get_web(self):
		return self.web
	def get_comment(self):
		return self.comment
	def get_creditDays(self):
		return self.creditDays
	def get_orgNo(self):
		return self.orgNo
	def get_preferredShipment(self):
		return self.preferredShipment
	def get_attachPdf(self):
		return self.attachPdf

	def set_name(self, localname):
		self.name = localname
	def set_zipcode(self, localzip):
		self.zipcode = localzip
	def set_city(self, localcity):
		self.city = localcity
	def set_address1(self, localaddr1):
		self.address1 = localaddr1
        def set_address2(self, localaddr2):
                self.address2 = localaddr2
	def set_country(self, localcountry):
		self.country = localcountry
	def set_recipientNo(self, localrecipientNo):
		self.recipientNo = localrecipientNo
        def set_phone(self, localphone):
		self.phone = localphone
        def set_mobile(self, localmobile):
		self.mobile = localmobile
        def set_fax(self, localfax):
		self.fax = localfax
        def set_email(self, localemail):
		self.email = localemail
        def set_web(self, localweb):
		self.web = localweb
	def set_comment(self, localcomment):
		self.comment = localcomment
        def set_creditDays(self, localcredit):
		self.creditDays = localcredit
        def set_orgNo(self, localorgNo):
		self.orgNo = localorgNo
        def set_preferredShipment(self, localpreferred):
		self.preferredShipment = localpreferred
        def set_attachPdf(self, localpdf):
		self.attachPdf = localpdf
			   				
